### DISCUSSION TITLE
Can't make Copilot work at all in Visual Studio

### DISCUSSION THREAD

**Discussion by ryobg** [community/community#70517] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Past two hours I have been trying to make Copilot 1.116 work in my VS Community 17.7.5. 

* I have paid subscription active for Copilot in my Github account
* The alternative extension for VSCode works
* I have tried reinstallation of the VS extension several times
* I have tried to sign in and out, adding and removing my GitHub account in VS
* I have tried and the https://github.com/login/device/ approach
* I can't open the Copilot (ctrl+alt+enter) due to "Exception has been thrown by the target of the invocation" error - there is no more info.
* I can't see any kind of suggestion from Copilot
* I have R#, toggled the suggested compatibility options without effect
* IntelliSense is disabled.
* Copilot always start with a disabled icon, despite being checked as Enabled
* Re-enabling again Copilot seems to work?

```
06:42:33.738 [Verbose] IsGloballyEnabled SetValue to True
06:42:33.739 [Verbose] AdviseSolutionEvents(VisualStudioEvents)
06:42:33.739 [Verbose] AdviseDebuggerEvents(VisualStudioEvents)
06:42:33.739 GitHub Copilot is globally enabled
06:42:33.751 [Verbose] TryInitializeAsync => StartAsync is null
06:42:33.780 [Verbose] OnLoadedAsync
06:42:33.783 [Verbose] ActivateAsync
06:42:33.789 [Verbose] LSP [Activated (36806163)]
06:42:33.789 [Verbose] LSP [Debug Server: http://localhost:8180]
06:42:33.789 [Verbose] LSP [Process: 34112]
06:42:33.794 [Verbose] AttachForCustomMessageAsync
06:42:33.806 [Verbose] LSP [file:///C:/Users/X/AppData/Local/Temp/VSLogs/VS32263860.LSPClient.GitHub.Copilot.CopilotLanguageClient.XKUG.svclog]
06:42:33.806 [Verbose] LSP [file:///C:/Users/X/AppData/Local/Temp/VisualStudio/LSP/CopilotLanguageClient-36c41421-f216-4f45-8d26-de94f1ef812d.log]
06:42:33.945 [DEBUG] [agent] [2023-10-19T03:42:33.939Z] Agent service starting
06:42:33.975 [Verbose] OnServerInitializedAsync
06:42:33.991 [Verbose] stderr:  [ApplicationInsights:An invalid instrumentation key was provided. There may be resulting telemetry loss [ undefined ]]
06:42:33.993 [DEBUG] [agent] [2023-10-19T03:42:33.991Z] Telemetry initialized
06:42:34.440 Auth Status: OK
06:42:34.491 [Verbose] JsonRpc.NotifyWithParameterObjectAsync: 'vs/didAddWorkspaceFolder'
06:42:34.495 [Verbose] _autoDetectSignInFlow: False
06:42:34.507 Successfully authorized GitHub Copilot.
06:42:34.507 [Verbose] [DeviceFlow] Current Auth Status SignInResult { Status = OK, Cancelled = False }
```

**Comment by ayim** [comment]
Hi @ryobg, I'm looking through the logs here and it looks like you're trying to get Copilot autocomplete to work (since this is the only one that supports device login). The logs, however, look like Copilot Chat.

Given the complexity, we'd love to hop on a call with you to get this fixed ASAP - please reach out at ayim@github.com or at aayim@microsoft.com.

**Reply by ryobg** [reply]
Thank you for reaching out ayim! I'm sending an email to the github account. The log is out of the "Github Copilot" console tab visible in Visual Studio. I'm trying to use only the Autocomplete feature. As an AI chat add-on, I have only the JetBrain's AI Assistant, but I think it is part of their extension.

**Reply by ayim** [reply]
Replied!

**Reply by ryobg** [reply]
I have seen that the VS ticket has been solved and I got an email notification from someone that a fix has been released (that message got deleted from here though). When I should verify that the solution works?

**Reply by ayim** [reply]
You should be good to go. If it's STILL crashing, you may need to manually download the latest package from the link below and install it.
https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

**Reply by ryobg** [reply]
Thanks for the heads up! However, I downloaded 1.139 and even updated the VS to 17.8.2 and the behaviour is still the same - there is no any kind of autocompletion suggestion from Copilot. I disabled/enabled through the icon just in any case, and still nothing.

**Reply by ayim** [reply]
If you're onlline now, I'll be online for the next 4 hours. Shoot me an email at aayim@microsoft.com and let's hop on a screenshare.

**Reply by sam-michel** [reply]
I am experiencing the exact same issue. Works in VS Code, not in Visual Studio Enterprise v17.7.7. Copilot always starts off disable (with a slash through it), but when I uncheck then recheck 'Enable Globally' it authenticates successfully and appears to be enable:

```
GitHub Copilot 1.139.0.0 (v1.139.0.0@4eca0246f)
GitHub Copilot Agent 1.139.0 (1.139.0)
GitHub Copilot is globally disabled
GitHub Copilot is globally enabled
[DEBUG] [agent] [2023-12-01T21:42:01.442Z] Agent service starting
[DEBUG] [agent] [2023-12-01T21:42:01.519Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.
```

I am getting no code completion suggestions and when I attempt to 'Open Copilot' I get the following error:

![image](https://github.com/community/community/assets/14353844/5e7474b9-0b3c-4da9-b4bf-d63a47005f91)

Any help on this would be greatly appreciated!

**Reply by ayim** [reply]
@sam-michel file a feedback ticket using the mechanism in screenshot below so we can get logs... and shoot me an email at aayim@microsoft.com

![image](https://github.com/community/community/assets/10951869/3cbfd7f0-e2aa-476f-96df-b68d19f84943)

**Comment by shyakachaste** [comment]
Despite your efforts to resolve issues with GitHub Copilot in Visual Studio Community 2017, encountering the "Exception has been thrown by the target of the invocation" error remains problematic. You've diligently attempted solutions, including reinstallation, signing in and out, and adjusting compatibility options. To further address the problem, ensure GitHub Copilot is compatible with your Visual Studio version, consider updating Visual Studio, review official documentation for known issues, contact GitHub support for personalized assistance, explore community forums for shared experiences, and, if needed, contemplate a fresh reinstall of both Copilot and Visual Studio. Providing the detailed error logs to GitHub support or community forums may offer more specific insights into resolving this challenging issue.

**Reply by ayim** [reply]
Hey @shyakachaste to use GitHub Copilot in Visual Studio, install Visual Studio 2022 version 17.6 or above.

**Comment by ryobg** [comment, marked as answer]
After a meetup with the support (thank you guys!):

1. We "Repair"-ed VS from from its Installer and made sure IntelliCode is installed.
2. IntelliCode throw up an error after the repair  (a support ticket was created for that).
3. I installed back my ReSharper and that fixed the IntelliCode error
4. I installed CoPilot again and voila - it is working!

CoPilot icon was enabled now on VS start, before it was not and I had to reset it. To me, it seems VS needed a Repair and I'm not sure that this IntelliCode depedency is working fluently. I have no idea how & whether not using ReSharper can make CoPilot work.

**Reply by sam-michel** [reply]
I am also using ReSharper and I suspected that might be the issue. I suspended ReSharper in ReSharper settings and disabled the ReSharper extension and restarted but I was still seeing the same exception on 'Open Copilot'.

I will try the repair/reinstall steps you shared above and see if that works for me as well, thanks!

**Reply by ayim** [reply]
@sam-michel when we were on a call with this customer, the big giveaway to this specific symptom was that there was no purple IntelliCode lightbulb icon next to the white Copilot icon in the bottom left of the code editor typing surface.

If you continue to have problems, please reach out to me at aayim@microsoft.com so we can diagnose together.

**Reply by sam-michel** [reply]
@ayim I was able to get Copilot working in VS Enterprise by essentially following the steps outlined above though I did them in a slightly different order:
1. Update VS 2022 to latest version (from 17.7.7 to 17.8.2)
2. Repair VS 2022 (I confirmed that IntelliCode was selected)
3. Install Copilot extension in VS 2022 (I did this first so I could confirm that it was working in a clean install)
4. Install ReSharper extension

I am now seeing the purple IntelliCode icon next to the Copilot icon (I wasn't seeing that before). Thanks for all your help!
